### PR TITLE
Initialize Vmec from a wout file without vmec or mpi 

### PR DESCRIPTION
**Pull request by ejpaul**
I made a few small tweaks to ``Vmec`` so that the class can be initialized from a ``wout`` file without importing ``vmec`` or ``MPI``. Since ``indata`` cannot be initialized from ``vmec.vmec_input``, I used the default values from ``vmec.vmec_input`` to initialize ``dofs``. Checks have been added to see if ``runnable = True`` before calling ``vmec.run()``.

Re-opened this PR (from #183) since I re-named the branch.

### CODE REVIEW

**Comment by landreman** [comment]
Can you refactor so all those extra `if runnable:` statements are not needed? In master, a `Vmec` object initialized from a `wout` file has `need_to_run_code=False` so `run()` just returns immediately, hence `if runnable:` shouldn't be needed. [These lines that were removed](https://github.com/hiddenSymmetries/simsopt/blob/master/src/simsopt/mhd/vmec.py#L286-L289) would have broken this, right? In fact I'd eventually like to get rid of the need to call `run()` entirely (this could be a separate PR), perhaps by renaming `.wout` to `._wout` and making `.wout` a property that calls `run()` and then returns `._wout`.

**Comment by ejpaul** [comment]
> Can you refactor so all those extra `if runnable:` statements are not needed? In master, a `Vmec` object initialized from a `wout` file has `need_to_run_code=False` so `run()` just returns immediately, hence `if runnable:` shouldn't be needed. [These lines that were removed](https://github.com/hiddenSymmetries/simsopt/blob/master/src/simsopt/mhd/vmec.py#L286-L289) would have broken this, right? In fact I'd eventually like to get rid of the need to call `run()` entirely (this could be a separate PR), perhaps by renaming `.wout` to `._wout` and making `.wout` a property that calls `run()` and then returns `._wout`.

The `if runnable` lines have been deleted in most places. I have kept this in when `need_to_run_code=True` is set in `vmec_diagnostics.py`.

**Review by landreman — COMMENTED** [review]
Here is an idea for extra test coverage. Presently there are some tests like `test_vacuum_well` and `test_iota` in `test_vmec.py`, and `test_independent_of_resolution` and `test_profile` in `test_vmec_diagnostics.py`, that are presently only run if vmec is installed, but they only use wout files so they should now work also if vmec is not installed. Would it work to move those tests to a new class outside the `unittest.skipIf` decorators?

**Comment by ejpaul** [comment]
> Here is an idea for extra test coverage. Presently there are some tests like `test_vacuum_well` and `test_iota` in `test_vmec.py`, and `test_independent_of_resolution` and `test_profile` in `test_vmec_diagnostics.py`, that are presently only run if vmec is installed, but they only use wout files so they should now work also if vmec is not installed. Would it work to move those tests to a new class outside the `unittest.skipIf` decorators?

I have moved the tests that are initialized from a wout file to the `InitializedFromWout` class in `test_vmec.py` and `test_vmec_diagnostics.py`.

File: src/simsopt/mhd/boozer.py [hunk 1]
```diff
@@ -127,7 +127,8 @@ def run(self):
         logger.info("Preparing to run Boozer transformation. Registry:{}".format(s))
 
         if isinstance(self.equil, Vmec):
-            self.equil.run()
+            if self.equil.runnable:
```
[the remark this answers is not in the document]
Comment (ejpaul) [reply src/simsopt/mhd/boozer.py]: I have removed these lines as suggested

File: src/simsopt/mhd/vmec_diagnostics.py [hunk 2]
```diff
@@ -461,8 +462,9 @@ def J(self):
         """
         Computes the quantity :math:`J` described in the class definition.
         """
-        self.vmec.need_to_run_code = True
-        self.vmec.run()
+        if (self.vmec.runnable):
```
[the remark this answers is not in the document]
Comment (ejpaul) [reply src/simsopt/mhd/vmec_diagnostics.py]: I have removed these lines as suggested

File: src/simsopt/mhd/vmec_diagnostics.py [hunk 3]
```diff
@@ -580,9 +582,9 @@ def J(self):
         """
         Computes the quantity :math:`J` described in the class definition.
         """
-
-        self.vmec.need_to_run_code = True
-        self.vmec.run()
+        if self.vmec.runnable:
```
[the remark this answers is not in the document]
Comment (ejpaul) [reply src/simsopt/mhd/vmec_diagnostics.py]: I have removed these lines as suggested

**Review by landreman — APPROVED** [review]
Looks good, thanks, go ahead and merge if you're ready.

**Pull request merged**